commons-hpcc/src/main/java/org/hpccsystems/commons/benchmarking/BenchmarkResult.java

```diff
@@ -0,0 +1,184 @@
+    {
+        testGroup = group;
+        testName = name;
+        testDescription = description;
+    }
+
+    public AveragedMetric getMetric(String name)
+    {
+        return metrics.get(name);
+    }
+
+    public void addMetric(AveragedMetric metric)
+    {
+        metrics.put(metric.name,metric);
+    }
+
+    public void addParameter(BenchmarkParam param)
+    {
+        parameters.add(param);
+    }
+
+    public JSONObject toJson()
+    {
+        return toJson(null);
+    }
+
+    private JSONObject serializeMetric(AveragedMetric metric, boolean flat)
+    {
+        if (flat)
+        {
+            JSONObject obj = new JSONObject();
+            obj.put("group",testGroup);
+            obj.put("test", testName);
+            obj.put("result", metric.name);
+            obj.put("value", metric.getAvg());
+
+            if (metric.units != null)
+            {
+                obj.put("units", metric.units);
+            }
+            
+            if (metric.description != null)
+            {
+                obj.put("description", metric.description);
+            }
+            return obj;
+        }
+        else
+        {
+            JSONObject obj = new JSONObject();
+            obj.put("name", metric.name);
+            
+            if (metric.units != null)
+            {
+                obj.put("units",metric.units);
+            }
+            
+            if (metric.description != null)
+            {
+                obj.put("description",metric.description);
+            }
+
+            obj.put("value",metric.getAvg());
+            return obj;
+        }
+    }
+
+    public JSONObject toJson(String[] selectedResults)
+    {
+        JSONObject test = new JSONObject();
+        test.put("name",this.testName);
+        test.put("description",this.testDescription);
+
+        // Write test params
+        JSONArray jsonParams = new JSONArray();
+
+        for (int i = 0; i < parameters.size(); i++)
+        {
+            JSONObject param = parameters.get(i).toJson();
+            jsonParams.put(param);
+        }
+
+        test.put("parameters",parameters);
+
+        // Write test results
+        JSONArray jsonMetrics = new JSONArray();
+
+        if (selectedResults == null)
+        {
+            for (HashMap.Entry<String, AveragedMetric> entry : metrics.entrySet())
+            {
+                AveragedMetric metric = entry.getValue();
+                metric.discardOutliers();
+                
+                JSONObject obj = serializeMetric(metric,false);
+                jsonMetrics.put(obj);
+            }
+        }
+        else
+        {
+            for (int i = 0; i < selectedResults.length; i++)
+            {
+                AveragedMetric metric = metrics.get(selectedResults[i]);
+                if (metric == null)
+                {
+                    continue;
+                }
+
+                metric.discardOutliers();
```

Comment:
are there any reasonable scenarios where a user wouldn't want to discard outliers as determined by the THOMPSON_TAU logic?

Reply:
Good question. If we are reporting the average then probably not, outliers can skew the average significantly.